src/main/java/org/isf/menu/rest/UserController.java

```diff
@@ -0,0 +1,292 @@
+	public ResponseEntity<List<UserGroupDTO>> getUserGroup() throws OHServiceException {
+		logger.info("Attempting to fetch the list of user groups");
+        List<UserGroup> groups = userManager.getUserGroup();
+        List<UserGroupDTO> mappedGroups = userGroupMapper.map2DTOList(groups);
+        if(mappedGroups.isEmpty()) {
+			logger.info("No group found");
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(mappedGroups);
+		} else {
+			logger.info("Found " + mappedGroups.size() + " groups");
+			return ResponseEntity.ok(mappedGroups);
+		}
+	}
+	
+	/**
+	 * returns the list of {@link UserMenuItem}s that compose the menu for specified {@link User}
+	 * @param userDTO - the {@link User}
+	 * @return the list of {@link UserMenuItem}s 
+	 */
+	@GetMapping(value = "/users/menus/{username}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<UserMenuItemDTO>> getMenu(@PathVariable String username) throws OHServiceException {
+		User user = userManager.getUserByName(username);
+		if(user == null) {
+			throw new OHAPIException(new OHExceptionMessage(null, "User not found!", OHSeverityLevel.ERROR));
+		}
+        List<UserMenuItem> menuItems = userManager.getMenu(user);
+        List<UserMenuItemDTO> mappedMenuItems = userMenuItemMapper.map2DTOList(menuItems);
+        if(mappedMenuItems.isEmpty()) {
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(mappedMenuItems);
+		} else {
+			return ResponseEntity.ok(mappedMenuItems);
+		}
+	}
+	
+	/**
+	 * returns the list of {@link UserMenuItem}s that compose the menu for specified {@link UserGroup}
+	 * @param aGroup - the {@link UserGroup}
+	 * @return the list of {@link UserMenuItem}s 
+	 */
+	@GetMapping(value = "/users/group-menus/{group_code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<UserMenuItemDTO>> getGroupMenu(@PathVariable("group_code") String code) throws OHServiceException {
+        UserGroup group = loadUserGroup(code);
+        List<UserMenuItemDTO> menus = userMenuItemMapper.map2DTOList(userManager.getGroupMenu(group));
+        if(menus.isEmpty()) {
+			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(menus);
+		} else {
+			return ResponseEntity.ok(menus);
+		}
+	}
+	
+	/**
+	 * replaces the {@link UserGroup} rights
+	 * @param code - the {@link UserGroup}'s code
+	 * @param menusDTO - the list of {@link UserMenuItem}s
+	 * @return <code>true</code> if the menu has been replaced, <code>false</code> otherwise.
+	 */
+	@PostMapping(value = "/users/groups/{group_code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Boolean> setGroupMenu(
+			@PathVariable("group_code") String code, 
+			@Valid @RequestBody List<UserMenuItemDTO> menusDTO) throws OHServiceException {
+		UserGroup group = loadUserGroup(code);
+        ArrayList<UserMenuItem> menus = new ArrayList<UserMenuItem>();
+        userMenuItemMapper.map2ModelList(menusDTO).forEach(m -> menus.add(m));
```

Comment:
We probably avoid foreach iteration assigning userMenuItemMapper.map2ModelList(menusDTO) to menus?